### PR TITLE
PHP 8, dropped doctrine/persistence < 1.3, static analysis

### PR DESCRIPTION
**Pull request by chives**

### CODE REVIEW

**Comment by chives** [comment]
@fsi-open

File: composer.json [hunk 1]
```diff
@@ -15,16 +15,17 @@
         }
     ],
     "require": {
-        "php": ">=7.1",
-        "symfony/event-dispatcher": "^2.2|^3.0|^4.0",
-        "symfony/options-resolver": "^2.6|^3.0|^4.0",
-        "fsi/data-indexer" : "0.9.*"
+        "php": "^7.1|^8.0",
+        "symfony/event-dispatcher": "^3.4|^4.0",
```
Comment (szymach) [inline composer.json:19]: Did not think of this before, but shouldn't we require Symfony components in versions `4.4.x` as well? Less dependency resolving and we don't use lower minor versions anyway.

File: lib/DataSourceFactory.php [hunk 2]
```diff
@@ -164,7 +164,7 @@ private function checkDataSourceName($name, DataSourceInterface $datasource = nu
             throw new DataSourceException('Name of data source must be unique.');
         }
 
-        if (!preg_match('/^[\w\d]+$/', $name)) {
+        if (!preg_match('/^[\w]+$/', $name)) {
```
Comment (szymach) [inline lib/DataSourceFactory.php:167]: `if (1 !== preg_match('/^[\w]+$/', $name)) {`?

File: lib/Driver/Collection/CollectionAbstractField.php [hunk 3]
```diff
@@ -45,14 +46,14 @@ public function buildCriteria(Criteria $c)
...
                 throw new CollectionDriverException(
                     'Fields with \'between\' comparison require to bind an array.'
                 );
             }
 
-            $from = count($data) ? array_shift($data) : null;
+            $from = array_shift($data);
             $to = count($data) ? array_shift($data) : null;
 
             if (!$from && ($from !== 0)) {
```
Comment (szymach) [inline lib/Driver/Collection/CollectionAbstractField.php:59]: When using `!`, isn't 0 included in it as well? I think the second part of the condition may be redundant.

File: composer.json (hunk 1 above)
Comment (chives) [reply composer.json:19]: maybe in next round of updates

File: lib/DataSourceFactory.php (hunk 2 above)
Comment (chives) [reply lib/DataSourceFactory.php:167]: srsly?

Comment (szymach) [reply lib/DataSourceFactory.php:167]: Yeah you're doing minor cs fixes anyway.

Comment (chives) [reply lib/DataSourceFactory.php:167]: but not everywhere

**Comment by chives** [comment]
@fsi-open

**Review by rn0 — APPROVED** [review]

**Review by szymach — APPROVED** [review]

**Pull request merged**